scripts/test_parse_directory.py

```diff
@@ -202,21 +208,29 @@ def main() -> None:
             f"or {total_bytes / total_seconds :,.0f} bytes/sec.",
         )
 
-    if args.short:
+    if short:
         print_memstats()
 
     if errors:
         print(f"Encountered {errors} failures.", file=sys.stderr)
 
     # Compare trees (the dict is empty unless -t is given)
     for file, tree in trees.items():
-        if not args.short:
+        if not short:
             print("Comparing ASTs for", file)
-        compare_trees(tree, file, verbose, args.tree >= 2)
+        compare_trees(tree, file, verbose, tree_arg >= 2)
 
     if errors:
         sys.exit(1)
 
 
 if __name__ == "__main__":
-    main()
+    args = argparser.parse_args()
+    directory = args.directory
+    grammar_file = args.grammar_file
+    verbose = args.verbose
+    excluded_files = args.exclude
+    skip_actions = args.skip_actions
+    tree = args.tree
+    short = args.short
+    main(directory, grammar_file, verbose, excluded_files, skip_actions, tree, short, None)
```

Comment:
I propose to rename main() to something else and put this code in a new argument-less main() function.